### PR TITLE
Pandas total step

### PR DESCRIPTION
**Pull request by luc-leonard**

### CODE REVIEW

**Comment by luc-leonard** [comment]
[(2) VQB pandas: totals step](https://trello.com/c/jtAwtBm0/1314-2-vqb-pandas-totals-step)

**Comment by fspot** [comment]
@luc-leonard I think you don't need to re-implement all the aggregation stuff, you can reuse the AggregateStep for that (see what I did in RollupStep for example).

File: server/weaverbird/steps/totals.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+class TotalsStep(BaseStep):
+    name = Field('totals', const=True)
+    total_dimensions: List[TotalDimension] = Field(alias='totalDimensions')
+    aggregations: List[Aggregation] = Field(min_items=1)
+    groups: List[ColumnName] = Field(min_items=0)
+
+    class Config:
+        allow_population_by_field_name = True
```
Comment (davinov) [inline server/weaverbird/steps/totals.py:23]: We could put this into the BaseStep ? (if that propagates, I'm not sure of this ^^)

File: server/weaverbird/steps/totals.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            for group_column in self.total_dimensions
+            if group_column != total_dimension
+        ]
+        aggregation = AggregateStep(
+            name='aggregate',
+            keepOriginalGranularity=False,
+            aggregations=self.aggregations,
+            on=group_by_columns,
+        )
+        df_result = aggregation.execute(df)
```
Comment (davinov) [inline server/weaverbird/steps/totals.py:80]: :100: super easy by reusing the other step :+1: 
Also it clearly show the "relation" between the two

**Review by davinov — APPROVED** [review]
Great! thx

**Pull request merged**